Fix tied params with Fake Tensor

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85065

Tested locally to fix `USE_FAKE_TENSOR=1 python benchmarks/huggingface.py --ci -d cuda --float32 --backend=aot_nop --training --only=RobertaForCausalLM`. When I tried to repro with a small example test could not successfully, but @anijain2305 is attempting to with minifier.

eellison committed:
Fix tied params with Fake Tensor

[ghstack-poisoned]

eellison committed:
Update on "Fix tied params with Fake Tensor"




[ghstack-poisoned]

eellison committed:
Update on "Fix tied params with Fake Tensor"


Tested locally to fix `USE_FAKE_TENSOR=1 python benchmarks/huggingface.py --ci -d cuda --float32 --backend=aot_nop --training --only=RobertaForCausalLM`. When I tried to repro with a small example test could not successfully, but anijain2305 is attempting to with minifier. 




[ghstack-poisoned]

eellison commented:
Filed issue here https://github.com/pytorch/pytorch/pull/85065

anijain2305 reviewed (approved):
LGTM, though I dont fully understand the details :|

eellison committed:
Update on "Fix tied params with Fake Tensor"


Tested locally to fix `USE_FAKE_TENSOR=1 python benchmarks/huggingface.py --ci -d cuda --float32 --backend=aot_nop --training --only=RobertaForCausalLM`. When I tried to repro with a small example test could not successfully, but anijain2305 is attempting to with minifier. 




[ghstack-poisoned]

eellison commented:
@pytorchbot merge
